worker/main.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"cloud.google.com/go/pubsub"
+	"context"
+	log "github.com/sirupsen/logrus"
+	"os"
+	"strconv"
+)
+
+type CSVWorker struct {
+	sample []byte
+}
+
+func init() {
+	// Only log the warning severity or above.
+	if "DEBUG" == getEnv("LOG_LEVEL", "DEBUG") {
+		log.SetLevel(log.DebugLevel)
+	} else {
+		log.SetLevel(log.InfoLevel)
+	}
+}
+
+func (cw CSVWorker) start() {
+	log.Debug("starting worker process")
+	ctx := context.Background()
+	client, err := pubsub.NewClient(ctx, getEnv("GOOGLE_CLOUD_PROJECT", "rm-ras-sandbox"))
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Close()
+	log.Debug("about to subscribe")
+	cw.subscribe(ctx, client)
+}
+
+func (cw CSVWorker) subscribe(ctx context.Context, client *pubsub.Client) {
+	subId := getEnv("PUBSUB_SUB_ID", "sample-workers")
+	log.WithField("subId", subId).Info("subscribing to subscription")
+	sub := client.Subscription(subId)
+	cctx, cancel := context.WithCancel(ctx)
+	log.Debug("waiting to receive")
+	err := sub.Receive(cctx, func(ctx context.Context, msg *pubsub.Message) {
+		log.Info("sample received - processing")
+		log.WithField("data", string(msg.Data)).Debug("sample data")
+		cw.sample = msg.Data
+		err := processSample(cw.sample)
+		if err != nil {
+			log.WithError(err).Error("error processing sample - nacking message")
+			msg.Nack()
+		} else {
+			log.Info("sample processed - acking message")
+			msg.Ack()
+		}
+	})
+
+	if err != nil {
+		log.WithError(err).Error("error subscribing")
+		cancel()
+	}
+}
+
+func getEnv(key string, defaultVar string) string {
```

Comment:
If you want something to do this in library form, I can heartily recommend `viper`

eg: https://github.com/ONSdigital/ras-rm-party/blob/master/config.go
https://github.com/ONSdigital/ras-rm-party/blob/master/main.go#L73
https://github.com/ONSdigital/ras-rm-party/blob/master/main.go#L79

Might save having to write this method yourself, and would mean not having to do the `strconv.Atoi` below because you could just use `viper.GetInt` instead

Reply:
Cool thanks, will take look